code/game/gamemodes/clock_cult/clock_objects.dm

```diff
@@ -0,0 +1,1568 @@
+/////////////////////
+// CLOCKWORK ITEMS //
+/////////////////////
+
+/obj/item/clockwork
+	name = "meme blaster"
+	desc = "What the fuck is this? It looks kinda like a frog."
+	var/clockwork_desc = "A fabled artifact from beyond the stars. Contains concentrated meme essence." //Shown to clockwork cultists instead of the normal description
+	icon = 'icons/obj/clockwork_objects.dmi'
+	icon_state = "rare_pepe"
+	w_class = 2
+
+/obj/item/clockwork/New()
+	..()
+	all_clockwork_objects += src
+
+/obj/item/clockwork/Destroy()
+	all_clockwork_objects -= src
+	return ..()
+
+/obj/item/clockwork/examine(mob/user)
+	if((is_servant_of_ratvar(user) || isobserver(user)) && clockwork_desc)
+		desc = clockwork_desc
+	..()
+	desc = initial(desc)
+
+/obj/item/clockwork/slab //Clockwork slab: The most important tool in Ratvar's arsenal. Allows scripture recital, tutorials, and generates components.
+	name = "clockwork slab"
+	desc = "A strange metal tablet. A clock in the center turns around and around."
+	clockwork_desc = "A link between the Celestial Derelict and the mortal plane. Contains limitless knowledge, fabricates components, and outputs a stream of information that only a trained eye can detect."
+	icon_state = "dread_ipad"
+	w_class = 3
+	var/list/stored_components = list("belligerent_eye" = 0, "vanguard_cogwheel" = 0, "guvax_capacitor" = 0, "replicant_alloy" = 0, "hierophant_ansible" = 0)
+	var/busy //If the slab is currently being used by something
+	var/production_cycle = 0
+	var/no_cost = FALSE //If the slab is admin-only and needs no components and has no scripture locks
+
+/obj/item/clockwork/slab/starter
+	stored_components = list("belligerent_eye" = 1, "vanguard_cogwheel" = 1, "guvax_capacitor" = 1, "replicant_alloy" = 1, "hierophant_ansible" = 1)
+
+/obj/item/clockwork/slab/debug
+	no_cost = TRUE
+
+/obj/item/clockwork/slab/debug/attack_hand(mob/living/user)
+	..()
+	if(!is_servant_of_ratvar(user))
+		add_servant_of_ratvar(user)
+
+/obj/item/clockwork/slab/New()
+	..()
+	SSobj.processing += src
+
+/obj/item/clockwork/slab/Destroy()
+	SSobj.processing -= src
+	..()
+
+/obj/item/clockwork/slab/process()
+	production_cycle++
```

Comment:
Ideally it should only work if it's in a mob, and if it is the only one in said mob.
Since the slabs only produce once every minute or so, a contents check right before production might work with not too much processing.